src/apps/myaccount/app.js

```diff
@@ -838,6 +854,49 @@ define(function(require) {
 					currentElement.val(currentElement.data('original_value'));
 				});
 			});
+
+			template.find('.change').on('click', function(e) {
+				e.preventDefault();
+
+				var currentElement = $(this),
+					module = currentElement.parents('#myaccount').find('.myaccount-menu .myaccount-element.active').data('module'),
+					moduleToUpdate = currentElement.data('module'),
+					fieldName = currentElement.data('field'),
+					newData = (function cleanFormData(moduleToUpdate, data) {
+						if (moduleToUpdate === 'billing') {
+							data.credit_card.expiration_date = data.extra.expiration_date.month + '/' + data.extra.expiration_date.year;
+						}
+
+						return data;
+					})(moduleToUpdate, monster.ui.getFormData('form_' + fieldName));
+
+				settingsValidate(fieldName, newData,
+					function() {
+						self.settingsUpdateData(moduleToUpdate, data[moduleToUpdate], newData,
+							function(data) {
+								var args = {
+									callback: function(parent) {
+										if (fieldName === 'credit_card') {
+											parent.find('.edition').hide();
+											parent.find('.uneditable').show();
+										} else if (fieldName === 'colorblind') {
```

Comment:
This block of code is no longer necessary, it will never be executed.
```suggestion
										if (fieldName === 'colorblind') {
```